### ISSUE TITLE
1Password and Google Authenticator cannot add otpauth:/ with "algorithm=SHA-1" in the TOTPAuthUri

### ISSUE THREAD

**Issue by dadonc** [epicweb-dev/totp#16]
Hi,
As the title says, both 1Password and Google Authenticator cannot add the generated otpauth:/ if it includes "algorithm=SHA-1".
(I have attached a screenshot of 1Password's error message.)
Using "algorithm=SHA1" it works.

Currently I am just overwriting the algorithm string (which is saved in the database as "SHA-1") when calling getTOTPAuthUri:

`  const otpUri = getTOTPAuthUri({
    ...verification,
    algorithm: "SHA1",
    accountName: user.email,
    issuer,
  });` 

With this change it works as expected.
I am writing this for documentation purposes but one could think about automatically doing this in getTOTPAuthUri or by changing the algorithm name to "SHA1" in the whole package.

Thanks for the package.
Cheers,
dadonc

<img width="1185" alt="Screenshot 2024-12-22 at 12 47 57" src="https://github.com/user-attachments/assets/009818a8-9bf9-4167-869f-65fa3e8dee52" />

**Comment by kentcdodds** [comment]
Yes, this is what I do: https://github.com/epicweb-dev/epic-stack/blob/2e177e8a685532d59ced66182b859126a5f4ec70/app/routes/settings%2B/profile.two-factor.verify.tsx#L73

The problem came from this: https://github.com/epicweb-dev/totp/pull/11

It would probably be a good idea to document this I think. Or perhaps we could make this a built-in feature of `getTOTPAuthUri`.

**Comment by kentcdodds** [comment]
I forgot to mention this issue when I fixed it in the commit, but this has now been fixed in [v3](https://github.com/epicweb-dev/totp/releases/tag/v3.0.0).

Thanks!
